src/huggingface_hub/_commit_api.py

```diff
@@ -449,9 +483,61 @@ def fetch_upload_modes(
     return upload_modes
 
 
+@validate_hf_hub_args
+def fetch_files_to_copy(
+    copies: Iterable[CommitOperationCopy],
+    repo_type: str,
+    repo_id: str,
+    token: Optional[str],
+    revision: str,
+    endpoint: Optional[str] = None,
+) -> Dict[Tuple[str, Optional[str]], "RepoFile"]:
```

Comment:
Thinking twice about it, I would refactor a bit this method in a `fetch_lfs_files_to_copy` helper. If we ever support copies of regular files, we'll reassess.

In particular:
- for each `CommitOperationCopy`, fetch the file details
    - if file doesn't exist: raise a `EntryNotFoundError("Cannot copy ...: file is missing on repo.")`
    - if file exist but not LFS: raise a `NotImplementedError("Cannot copy a regular file.")`
- return the `sha256` for each `filename + revision` pair

The goal of raising an exception early is to avoid uploading LFS files and then raise an exception (if commit with multiple operations).